src/connection.ts

```diff
@@ -269,13 +277,18 @@ export class Connection extends JSForceConnection {
       this.query<T>(soql, _options)
         .on('record', rec => records.push(rec))
         .on('error', err => reject(err))
-        .on('end', () =>
+        .on('end', () => {
+          if (records.length >= 10000 || limit < 10000) {
```

Comment:
This seems wrong to me because `records.length` is only ever the size of records that are returned. It seems like what you want is the `totalSize` set on the query. Also if `totalSize` is 2000 and limit is 6000, that means there are no missing records.

You probably want something like:

```
const query = this.query<T>(soql, _options);
query.on('record', rec => records.push(rec))
query.on('error', err => reject(err))
query.on('end', () =>
    if (query.totalSize > records.length) { // This should give you if there are missing records.
```

**Note** that I didn't use limit to determine if here are missing records. I believe query is in batches of 2k records. So if you have a limit there are 6000 records, and you set the limit to 5000, it **will** return all 6k records.